pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -384,13 +384,24 @@ func (r *ReconcileBareMetalHost) actionDeleting(prov provisioner.Provisioner, in
 		info.log.Info("ready to be deleted")
 		return deleteComplete{}
 	}
+	// try to power off host before delete
+	provResult, err := prov.PowerOff()
+	if err != nil {
+		return actionError{errors.Wrap(err, "failed to power off host before delete")}
+	}
+	if provResult.Dirty {
+		if err != nil {
+			return actionError{errors.Wrap(err, "failed to power off host before deleting")}
+		}
+		return actionContinue{provResult.RequeueAfter}
+	}
 
-	provResult, err := prov.Delete()
+	// proceed with the delete
+	provResult, err = prov.Delete()
 	if err != nil {
 		return actionError{errors.Wrap(err, "failed to delete")}
 	}
 	if provResult.Dirty {
-		err = r.saveStatus(info.host)
 		if err != nil {
```

Comment:
This is now unreachable also, since it was only here to check the result of saveStatus().